packages/sessions-sdk-ts/src/connection.ts

```diff
@@ -236,9 +241,11 @@ const buildTransaction = async (
   const [{ value: latestBlockhash }, sponsor, addressLookupTable, signers] =
     await Promise.all([
       connection.rpc.getLatestBlockhash().send(),
-      connection.sponsor === undefined
-        ? getSponsor(connection, connection.sponsorCache, domain)
-        : Promise.resolve(connection.sponsor),
+      sponsorOverride
+        ? Promise.resolve(sponsorOverride)
+        : (connection.sponsor === undefined
+          ? getSponsor(connection, connection.sponsorCache, domain)
+          : Promise.resolve(connection.sponsor)),
```

Comment:
```suggestion
      sopnsorOverride === undefined && connection.sponsor === undefined
        ? getSponsor(connection, connection.sponsorCache, domain)
        : Promise.resolve(sponsorOverride ?? connection.sponsor)),
```

minor nit but I really dislike the nested ternary